File: urbanairship-core/src/main/java/com/urbanairship/AirshipLoopers.java
package com.urbanairship;

import android.os.Looper;

import com.urbanairship.util.AirshipHandlerThread;

import androidx.annotation.NonNull;

/**
 * Shared SDK loopers.
 */
public class AirshipLoopers {

    private static Looper backgroundLooper;

    /**
     * Gets the background looper.
     *
     * @return The background looper.
     */
    @NonNull
    public static Looper getBackgroundLooper() {
        if (backgroundLooper == null) {
            synchronized (AirshipLoopers.class) {
                if (backgroundLooper == null) {
                    AirshipHandlerThread thread = new AirshipHandlerThread("background");
                    thread.start();
                    backgroundLooper = thread.getLooper();
                }
            }
        }
        return backgroundLooper;
    }

}

File: urbanairship-core/src/main/java/com/urbanairship/AirshipLoopers.kt
package com.urbanairship

import android.os.Looper
import com.urbanairship.util.AirshipHandlerThread

/**
 * Shared SDK loopers.
 */
public object AirshipLoopers {

    /**
     * Gets the background looper.
     */
    @JvmStatic
    public val backgroundLooper: Looper by lazy(mode = LazyThreadSafetyMode.SYNCHRONIZED) {
        val thread = AirshipHandlerThread("background")
        thread.start()
        return@lazy thread.looper
    }
}
